### PR TITLE
Signatures for aliases

### LINKED ISSUE #4457: Allow a signature per alias
### Feature Request

Currently I can pick one signature per account, but I'd like to also have the ability to select one per alias.

#### Summary

n/a

- [x] Requires https://github.com/nextcloud/mail/issues/1660

### PR DESCRIPTION
**Pull request by kesselb**
Fix #4457 

| > 1 identity = select | 1 identity = no select |
| --- | --- |
| Screenshot from 2021-04-06 21-30-32 | Screenshot from 2021-04-06 21-34-08 |

**Todo**

- [x] Edge case: Update currently open draft with new signature
- [x] Unit-Tests

### CODE REVIEW

File: src/service/AliasService.js [hunk 1]
```diff
@@ -26,3 +23,19 @@ export const deleteAlias = async(account, alias) => {
 
 	return axios.delete(url).then((resp) => resp.data)
 }
+
+class AliasService {
```
Comment (kesselb) [inline src/service/AliasService.js:27]: @ChristophWurst can we use classes?

**Comment by kesselb** [comment]
>  Edge case: Update currently open draft with new signature

Is more complex then expected. I tried to write a watcher for the signature property. The property is deeply nested in state.accounts.

File: src/service/AliasService.js (hunk 1 above)
Comment (ChristophWurst) [reply src/service/AliasService.js:27]: possible but typically not required. what would be the benefit here?

Comment (kesselb) [reply src/service/AliasService.js:27]: it's much nicer ;)

File: src/components/Composer.vue [hunk 2]
```diff
@@ -462,6 +461,21 @@ export default {
...
+		aliases(newAliases) {
+			console.debug('aliases changed')
+			if (this.selectedAlias === NO_ALIAS_SET) {
+				return
+			}
+
+			const newAlias = newAliases.find(alias => alias.id === this.selectedAlias.id && alias.aliasId === this.selectedAlias.aliasId)
+			if (newAlias === undefined) {
+				// selected alias does not exist anymore.
+				this.onAliasChange(newAliases[0])
```
Comment (kesselb) [inline src/components/Composer.vue:473]: I wonder if newAliases could be empty. We have the default "alias" from the email address.

Comment (ChristophWurst) [reply src/components/Composer.vue:473]: we can defensively add a check

File: src/components/Composer.vue [hunk 3]
```diff
@@ -462,6 +461,21 @@ export default {
 		allRecipients() {
 			this.checkRecipientsKeys()
 		},
+		aliases(newAliases) {
+			console.debug('aliases changed')
+			if (this.selectedAlias === NO_ALIAS_SET) {
+				return
+			}
+
+			const newAlias = newAliases.find(alias => alias.id === this.selectedAlias.id && alias.aliasId === this.selectedAlias.aliasId)
```
Comment (ChristophWurst) [inline src/components/Composer.vue:470]: :thinking: :thinking: :thinking: 

so ``alias.id`` is the *account id* and ``alias.aliasId`` is the *alias id*?

Comment (kesselb) [reply src/components/Composer.vue:470]: image

Yes

File: src/components/Composer.vue (hunk 2 above)
Comment (kesselb) [reply src/components/Composer.vue:473]: should be fine without. the account is always there.

File: src/components/Composer.vue (hunk 3 above)
Comment (ChristophWurst) [reply src/components/Composer.vue:470]: could we flip that so that ``alias.id`` is the *alias id?

and for the comparison it should be sufficient to compare the alias IDs as those are unique, right?

**Review by ChristophWurst — APPROVED** [review]
Smoooooooooooooth! Works super well :shipit: 

Let's do the account id vs alias id in another PR, ok?
